providers/gce/gce_instances.go

```diff
@@ -121,12 +151,17 @@ func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v
 				if err != nil {
 					return nil, fmt.Errorf("couldn't get internal IP: %v", err)
 				}
-				nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: internalIP})
+
+				if internalIP == "" {
+					klog.Infof("No internal IPv4 address found for node %v.", nodeName)
+				} else {
+					nodeAddresses = append(nodeAddresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: internalIP})
+				}
 
 				// Both internal and external IPv6 addresses are written to this array
 				ipv6s, err := metadata.Get(fmt.Sprintf(networkInterfaceIPV6, nic))
 				if err != nil || ipv6s == "" {
-					klog.Infof("no internal IPV6 addresses found for node %v: %v", nodeName, err)
+					klog.Infof("No internal IPV6 addresses found for node %v: %v.", nodeName, err)
```

Comment:
just an advice, you need to avoid modifying unrelated part of the code in the same commit

Reply:
These are tiny readability-related enhancements. I can revert them, though, and submit in a different PR. Please let me know.

Reply:
no, is not important, is for education, each commit has to be self contained, if you want to do unrelated changes you can use a different commit or a different PR, but if you make them in the same commit it is distracting for the reviewer and in case of reverts or bisect or cherry-pick it can have conflicts ...